src/domain/datasets/services/src/dataset_key_value_service_sys_env.rs

```diff
@@ -35,15 +36,28 @@ impl DatasetKeyValueServiceSysEnv {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#[async_trait::async_trait]
 impl DatasetKeyValueService for DatasetKeyValueServiceSysEnv {
-    async fn find_dataset_env_var_value_by_key<'a>(
+    fn find_dataset_env_var_value_by_key(
         &self,
         dataset_env_var_key: &str,
-        _dataset_env_vars: &'a HashMap<String, DatasetEnvVar>,
+        dataset_env_vars: &HashMap<String, DatasetEnvVar>,
     ) -> Result<DatasetEnvVarValue, FindDatasetEnvVarError> {
+        // Search explicitly provided vars first
+        if let Some(var) = dataset_env_vars.get(dataset_env_var_key) {
```

Comment:
This is just to simplify testing. I was hoping for a simpler way to pass predefined vars into the `FetchService`, but this was the simplest way without a major refactoring.